### PR TITLE
Bump dependencies and remove dask

### PR DESCRIPTION
**Pull request by phackstock**
- [ ] ~Tests added~
- [ ] ~Documentation added~
- [ ] ~Example added (in the documentation, to an existing notebook, or in a new notebook)~
- [x] Description in ``CHANGELOG.rst`` added (single line such as: ``(`#XX <https://github.com/iiasa/climate-assessment/pull/XX>`_) Added feature which does something``)

Tests run locally, seems the failing tests are due to an issue with downloading MAGICC. It seems unrelated to this PR https://github.com/iiasa/climate-assessment/actions/workflows/nightly.yaml. Looks like it started failing on the 20th of April 2024.

### CODE REVIEW

**Review by znicholls — DISMISSED** [review]
All fine. I'd prefer to stick to pins if we can.

Can you add a CHANGELOG entry too please, then good to go.

File: pyproject.toml [hunk 1]
```diff
@@ -35,14 +35,13 @@ homepage = "https://github.com/iiasa/climate-assessment"
...
 fair = "==1.6.2"
-ixmp4 = "==0.8.0"
+ixmp4 = ">=0.8.3"
 joblib = "==1.3.2"
 openscm-runner = "==0.12.1"
-pandas = "==2.1.2"
+pandas = ">=2.1.2"
 pooch = "==1.8.0"
-pyam-iamc = "==2.1.0"
+pyam-iamc = ">=2.2.1"
```
[the remark this answers is not in the document]
Comment (phackstock) [reply pyproject.toml:44]: Sounds good, I'll pin the versions to what we have and then merge.

**Review by znichollscr — APPROVED** [review]
Nice, thanks

**Review by znicholls — APPROVED** [review]
Review from the right account now 🙃

**Pull request merged**